hyperparameter_search.py

```diff
@@ -297,17 +298,17 @@ def objective(trial, MODEL_NAME, NUM_CLASSES, N_EPOCHS, OPTIMIZER_SEARCH_SPACE,
 
 @click.command()
 @click.option('-m', '--model', required=True, type=click.Choice(AVAILABLE_MODELS, case_sensitive=False), help='Model architechture.')
-@click.option('-e', '--no_of_epochs', type=int, default=20, help='Number of epochs in training loop.')
-@click.option('-t', '--no_of_trials', type=int, default=50, help='Number of hyperparamter search trials in training loop.')
+@click.option('-e', '--no_of_epochs', type=int, show_default=True, default=20, help='Number of epochs in training loop.')
+@click.option('-es', '--early_stopping_counter', type=int, help='Number of consequtive epochs with no improvement in loss until trial is stopped. Default: ~one third of no of epochs.')
```

Comment:
I would limit the maximum patience value to ~10 or so epochs. I would find it very unlikely that with 200 epochs the validation loss would decrease in 66 epochs consecutively.

```suggestion
@click.option('-es', '--early_stopping_counter', type=int, help='Number of consecutive epochs with no improvement in loss until trial is stopped. Default: ~one third of no of epochs or 10, which ever is smaller.')

```